### PR TITLE
CLOUDP-222246: Send CI reports via Slack

### PR DESCRIPTION
**Pull request by josvazg**
Added:
- Summary report format for the reports.
- Script to slack the reports.
- Makefile rules to create and slack the reports.
- A new workflow to call the makefile that computes & slacks the reports.

### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

**Comment by josvazg** [comment]
Unfortunately, I think I cannot test the new workflow until this merges and main is aware of it.

File: .github/workflows/reports.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI Reports
+
+on:
+  workflow_call:
+  schedule:
+    - cron: "0 0 * * 1" # At 0:00 on Monday
```
Comment (helderjs) [inline .github/workflows/reports.yml:6]: How long does it take to run? isn't it better to run it at 7 am instead?

**Review by helderjs — APPROVED** [review]
LGTM, just left a suggestion

File: .github/workflows/reports.yml (hunk 1 above)
Comment (josvazg) [reply .github/workflows/reports.yml:6]: The local runs were under 4 minutes for both. I guess 10 minutes tops on the CI?

Why is 7 AM better? we want the data from last week, which ends at Monday midnight

Comment (helderjs) [reply .github/workflows/reports.yml:6]: While the message would not tag anyone, sending a message to our channel in the middle of the night from Monday to Sunday doesn't sound a good idea.

Comment (josvazg) [inline .github/workflows/reports.yml:6]:
```suggestion
    - cron: "0 7 * * 1" # At 7:00 on Monday
```

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**